### ISSUE TITLE
CSS prop shows a TypeScript error for properties with undefined values

### ISSUE THREAD

**Issue by jorenrui** [modulz/stitches#375]
# Bug report

Repro: https://codesandbox.io/s/solitary-butterfly-udx18?file=/src/App.tsx

## Describe the bug

I was using stitches' alpha version, when I tried migrating to the beta version, I now received this error.

When you give a property value of `undefined` for a styled component's css prop, it shows a TypeScript error.

```jsx
<StyledComponent css={{ width: isMobile ? '10px' : undefined  }} />
```

## Expected behavior

It should work like how it was before in Alpha.

## Screenshots
![image](https://user-images.githubusercontent.com/23741509/107970771-86d4f380-6fec-11eb-850d-f35c02ca5eea.png)

![image](https://user-images.githubusercontent.com/23741509/107970008-8851ec00-6feb-11eb-9f89-03c77ad3ced6.png)

## System information

- OS: macOs
- Version of Stitches: 0.1.0-canary.3
- Version of Node.js: v14.15.4
- Version of TypeScript: 4.1.2

**Comment by andyhqtran** [comment]
Adding on to this, it seems like we get this type error when spreading `css` into the `css` prop.

```tsx
const StyledButton = styled('button', { ...stylesHere });

type StyledButtonProps = ComponentProps<typeof StyledButton>;

type BigButtonProps = StyledButtonsProps; 

const BigButton = ({ css }: BigButtonProps) => {
	return (
		<StyledButton
			css={{
				backgroundColor: 'transparent',
				// This causes a type error
				...(css ?? {}),
				// This also causes a type error
				...css
			}}
		/>
	);
};
```

**Comment by peduarte** [comment]
@andyhqtran I cant reproduce. This works fine for me
```
const Button = styled('button', {});

type ButtonProps = React.ComponentProps<typeof Button>;

function MyButton({ css, ...props }: ButtonProps) {
  return <Button {...props} css={{ ...css }} />;
}
```

![image](https://user-images.githubusercontent.com/372831/108095597-53dc3f80-7080-11eb-8248-8f22e5bfb46f.png)

**Comment by peduarte** [comment]
Fixed in #389

**Comment by peduarte** [comment]
@andyhqtran please raise a new issue with a repro if you could. That would be very helpful. Thanks
